frontend/app/routes/_public+/apply+/$id+/demographics-part2.tsx

```diff
@@ -19,81 +25,187 @@ export const handle = {
 };
 
 export async function loader({ request, params }: LoaderFunctionArgs) {
+  const { OTHER_GENDER_TYPE_ID } = getEnv();
   const sexAtBirthTypes = await getLookupService().getAllSexAtBirthTypes();
   const mouthPainTypes = await getLookupService().getAllMouthPaintTypes();
   const lastTimeDentistVisitTypes = await getLookupService().getAllLastTimeDentistVisitTypes();
   const avoidedDentalCostTypes = await getLookupService().getAllAvoidedDentalCostTypes();
+  const genderTypes = await getLookupService().getAllGenderTypes();
   const applyFlow = getApplyFlow();
   const { id, state } = await applyFlow.loadState({ request, params });
-  return json({ sexAtBirthTypes, mouthPainTypes, lastTimeDentistVisitTypes, avoidedDentalCostTypes, id, state });
+
+  const otherGenderCode = genderTypes.find((method) => method.id === OTHER_GENDER_TYPE_ID);
+  if (!otherGenderCode) {
+    throw new Response('Expected gender type!', { status: 500 });
+  }
+
+  return json({ otherGenderCode, genderTypes, sexAtBirthTypes, mouthPainTypes, lastTimeDentistVisitTypes, avoidedDentalCostTypes, id, state: state.demographicsPart2 });
 }
 
-export async function action({ request }: ActionFunctionArgs) {
-  return json('');
+export async function action({ request, params }: ActionFunctionArgs) {
+  const applyFlow = getApplyFlow();
+  const { id } = await applyFlow.loadState({ request, params });
+
+  const formSchema = z.object({
+    genderTypeChoosen: z.string({ required_error: 'empty-radio' }),
```

Comment:
```suggestion
    genderTypeChosen: z.string({ required_error: 'empty-radio' }),
```

Or if we remove the (unnecessary IMO) suffix:
```suggestion
    gender: z.string({ required_error: 'empty-radio' }),
```

Reply:
Renamed as requested